src/runtime_src/core/pcie/driver/linux/xocl/subdev/icap.c

```diff
@@ -1521,13 +1521,22 @@ static int icap_create_subdev_cu(struct platform_device *pdev)
 	for (i = 0; i < ip_layout->m_count; ++i) {
 		struct xocl_subdev_info subdev_info = XOCL_DEVINFO_CU;
 		struct ip_data *ip = &ip_layout->m_ip_data[i];
+		u32 range;
 
 		if (ip->m_type != IP_KERNEL)
 			continue;
 
 		if (ip->m_base_address == 0xFFFFFFFF)
 			continue;
 
+		range = subdev_info.res[0].end - subdev_info.res[0].start;
+		/* If CU's address is not aligned, remove CU's resource */
```

Comment:
i thought we want to use cu type not address, no?

Reply:
This is U30 branch specific. There would be no CU type change in the xclbin.
For master, I would provide a different change.